rfcs/2022-03-22-motion-rfc.md

```diff
@@ -0,0 +1,612 @@
+
+- We can use Bezier Curves with both **React** & **React Native** to define the easing of our animations.
+ 
+- **CSS** natively understands `cubic-bezier` functions and allows you to define `transition-timing-function: cubic-bezier(0, 0, 1, 1)`. Other libraries like `framer-motion`, `react-spring` & `react-motion` also allow you to define your easing with a `cubic-bezier` function.
+
+- React Native's **Animated API** understands `Easing` functions and allows you to define `easing: Easing.bezier(0, 0, 1, 1)`. Other libraries like `react-native-reanimated` also allow you to define your easing with their own `Easing.bezier` function.
+
+To understand easing better, lets take a look at a few different examples of easing:
+
+**Linear Easing:**
+
+If we move an object from one point to another with a linear motion where it's acceleration as well as deceleration is linear, it would look like this: 
+
+<img alt="linear-easing" src="./images/motion/linear-easing.gif" width="300px">
+
+We can represent this as a bezier function `cubic-bezier(0, 0, 1, 1)` and a bezier curve:
+
+<img alt="linear-curve" src="./images/motion/linear-curve.png" width="200px">
+
+**Rapid Start - Slow End Easing:**
+
+If we move an object from one point to another where it will start at a higher velocity and slow down as it approaches the destination, it would look like this:
+
+<img alt="rapid-start-easing" src="./images/motion/rapid-start-easing.gif" width="300px">
+
+We can represent this as a bezier function `cubic-bezier(0, 1, 1, 1)` and a bezier curve:
+
+<img alt="rapid-start-curve" src="./images/motion/rapid-start-curve.png" width="200px">
+
+**Ease In - Ease Out Easing:**
+
+If you're familiar with easing in CSS, you must have come across a transition timing property of `ease-in-out`. This is the same as linear but with a slower acceleration at the beginning and a slower deceleration at the end.
+
+<img alt="ease-in-out-easing" src="./images/motion/ease-in-out-easing.gif" width="300px">
+
+We can represent this as a bezier function `cubic-bezier(0.42, 0, 0.58, 1)` and a bezier curve:
+
+<img alt="ease-in-out-curve" src="./images/motion/ease-in-out-curve.png" width="200px">
+
+> You can experiment with different bezier curves to see how they look here: [cubic-bezier.com](https://cubic-bezier.com/)
+
+#### Tokens - Easing
+> Note: *The naming for these tokens is not finalized yet. We will be updating this in the future.*
+
+- We will be storing **Easing tokens** in `blade/src/tokens/global/motion.ts` as a `cubic-bezier()` function.
+
+- After thorough research & experimentation, our design team has created 3 broad **intents** of easing: 
+  - `standard`
+  -  `entrance`
+  -   `exit` 
```

Comment:
**Naming opinion:**

To me standard is a bit confusing, the intent isn't obvious when should I use. 

A better naming would be "idle" where it hints to the consumer that this token is used for objects which are idle and not entering or exiting. It conveys they intent much better IMO.

Reply:
`standard` is just the standard easing we'll be using for everything. `idle` might not be the case every time. We could go with `default` but that seems more from the sense that if you don't apply anything, this would be the default anyway. `standard` seems like an intent you'd have for all easings that aren't entering or exiting